modules/account-lib/src/coin/sol/transactionBuilder.ts

```diff
@@ -21,55 +38,210 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
+    }
+    this.transaction.loadInputsAndOutputs();
+    return this.transaction;
   }
 
   /**
    * Builds the solana transaction.
    */
-  protected abstract buildSolTransaction(): SolTransaction;
+  protected buildSolTransaction(): SolTransaction {
+    const tx = new SolTransaction();
+    if (this._transaction?.solTransaction?.signatures) {
+      tx.signatures = this._transaction?.solTransaction?.signatures;
+    }
+    assert(this._feePayer, new BuildTransactionError('feePayer is required before building'));
+    tx.feePayer = new PublicKey(this._feePayer);
+
+    if (this._nonceInfo) {
+      tx.nonceInfo = {
+        nonce: this._recentBlockhash,
+        nonceInstruction: solInstructionFactory(this._nonceInfo)[0],
+      };
+    } else {
+      tx.recentBlockhash = this._recentBlockhash;
+    }
+    for (const instruction of this._instructionsData) {
+      tx.add(...solInstructionFactory(instruction));
+    }
+    for (const signer of this._signers) {
+      const publicKey = new PublicKey(signer.getKeys().pub);
+      const secretKey = signer.getKeys(true).prv;
+      assert(secretKey instanceof Uint8Array);
+      tx.partialSign({ publicKey, secretKey });
+    }
+    return tx;
+  }
 
   // region Getters and Setters
   /** @inheritdoc */
   protected get transaction(): Transaction {
     return this._transaction;
   }
+
+  /** @inheritdoc */
+  protected set transaction(transaction: Transaction) {
+    this._transaction = transaction;
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    this.validateKey(key);
+    this.checkDuplicatedSigner(key);
+    const prv = key.key;
+    const signer = new KeyPair({ prv: prv });
+    this._signers.push(signer);
+
+    return this._transaction;
+  }
+
+  /**
+   * Set the transaction fee payer
+   *
+   * @param {string} feePayerAddress the fee payer account
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  feePayer(feePayerAddress: string | undefined): this {
+    if (!feePayerAddress || !isValidPublicKey(feePayerAddress)) {
+      throw new BuildTransactionError('Invalid or missing feePayerAddress, got: ' + feePayerAddress);
+    }
+    this._feePayer = feePayerAddress;
+    return this;
+  }
+
+  /**
+   * Set the transaction nonce
+   * Requires both optional params in order to use the durable nonce
+   *
+   * @param {Blockhash} blockHash The latest blockHash
+   * @param {boolean} useDurableNonce Set to true to use Durable Nonce, default is false
+   * @param {string} [walletNoncePubKey] the pub key of the nonce Wallet (required for durable nonce)
+   * @param {string} [authWalletPubKey] the pub key of the owner of the nonce Wallet (required for durable nonce)
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  nonce(blockHash: Blockhash, useDurableNonce = false, walletNoncePubKey?: string, authWalletPubKey?: string): this {
```

Comment:
looks like `walletNoncePubKey` and `authWalletPubKey` need to be provided together.
lets create a new interface for it, maybe something like:
```
interface DurableNonceParams {
  walletNonceAddress: string,
  authWalletAddress: string
}
```

then you can update function signature to something like:
```
nonce(blockHash: Blockhash, durableNonceParams?: DurableNonceParams)
```

note users should not need to set `useDurableNonce` explicitly. its derived from the existence of `durableNonceParams`

Reply:
also, if we use the concept where `feePayer === sender`, we don't need a separate `authWalletAddress`.

for the initial impl, let's assume that we will only support use cases where there is 1 signer.